presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PartialAggregationPushDown.java

```diff
@@ -146,31 +146,26 @@ private PlanNode pushPartial(AggregationNode partial, ExchangeNode exchange)
             for (int i = 0; i < exchange.getSources().size(); i++) {
                 PlanNode source = exchange.getSources().get(i);
 
-                if (!exchange.getOutputSymbols().equals(exchange.getInputs().get(i))) {
-                    // Add an identity projection to preserve the inputs to the aggregation, if necessary.
-                    // This allows us to avoid having to rewrite the symbols in the aggregation node below.
-                    Assignments.Builder assignments = Assignments.builder();
-                    for (int outputIndex = 0; outputIndex < exchange.getOutputSymbols().size(); outputIndex++) {
-                        Symbol output = exchange.getOutputSymbols().get(outputIndex);
-                        Symbol input = exchange.getInputs().get(i).get(outputIndex);
-                        assignments.put(output, input.toSymbolReference());
+                ImmutableMap.Builder<Symbol, Symbol> mappingsBuilder = ImmutableMap.builder();
```

Comment:
I have some problem with `SymbolMapper`. It seems to be somehow disabled (crooked). Though, it is going to be short-lived class, so I am not sure how much effort is worth to put in it.

Anyway, I think it would be nice to have `SymbolMapper#Builder` which you could use here to  create a `SymbolMapper`.